pkg/operator/ceph/cluster/osd/envs.go

```diff
@@ -205,11 +205,6 @@ func cephVolumeEnvVar() []v1.EnvVar {
 
 func osdActivateEnvVar() []v1.EnvVar {
 	monEnvVars := []v1.EnvVar{
-		{Name: "ROOK_CEPH_MON_HOST",
```

Comment:
This might not work as we are using the mon end points the rook-ceph config secret. 

One option to pass the ceph config as `CEPH_ARGS` instead of the mon endpoints,  that is, `-c <ceph.config>` instead of `-m <mon-endpoints`.

Reply:
passing in the config map location with `-c` in the CEPH_ARGS has worked me

``` 
// {Name: "ROOK_CEPH_MON_HOST",
		// 	ValueFrom: &v1.EnvVarSource{
		// 		SecretKeyRef: &v1.SecretKeySelector{LocalObjectReference: v1.LocalObjectReference{
		// 			Name: "rook-ceph-config"},
		// 			Key: "mon_host"}}},
		// {Name: "CEPH_ARGS", Value: "-m $(ROOK_CEPH_MON_HOST)"},
		{Name: "CEPH_ARGS", Value: fmt.Sprintf("-c %s/%s/%s.config", dataDir, c.clusterInfo.Namespace, c.clusterInfo.Namespace)},
```

Reply:
Thanks a lot @sp98 for testing it. Yes, for OSD pod it works with passing the ceph config, except for the ActivateOSD when it sets envVars in `getActivateOSDInitContainer(..)`